app/serializers/partner_deal_serializer.rb

```diff
@@ -1,12 +1,23 @@
 class PartnerDealSerializer < ActiveModel::Serializer
   attributes :id, :partner_name, :partner_logo, :deal_title, :deal_url,
-    :deal_description, :redemption_info, :recurring, :has_codes, :code
+    :deal_description, :redemption_info, :recurring, :has_codes, :code,
+    :claimed_at
 
   def has_codes
     object.codes.unclaimed.count > 0
   end
 
   def code
-    scope && object.codes.where(user: scope).last.try(:code)
+    code_object && code_object.code
+  end
+
+  def claimed_at
+    code_object && code_object.claimed_at
+  end
+
+  private
+
+  def code_object
+    scope && object.codes.where(user: scope).last
```

Comment:
This seems like a bad idea to use queries directly in a serializer.  Is there some other way this could be done?